core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataMetricsTest.java

```diff
@@ -0,0 +1,715 @@
+        DescendantType.NONE, ListStatusContext.defaults().getOptions().getCommonOptions(),
+        false, true, false, false);
+    syncStream.sync();
+
+    assertEquals(previousStreamCount + 1, streamCount.getCount());
+    previousStreamCount += 1;
+    assertEquals(previousSucceededStreams + 1, succeededStreams.getCount());
+    previousSucceededStreams += 1;
+    assertEquals(previousSucceededPaths + 1, succeededPaths.getCount());
+    previousSucceededPaths += 1;
+    assertEquals(previousFailedStreams + 0, failedStreams.getCount());
+    previousFailedStreams += 0;
+    assertEquals(previousFailedPaths + 0, failedPaths.getCount());
+    previousFailedPaths += 0;
+    assertEquals(previousNoChangedPaths + 1, noChangedPaths.getCount());
+    previousNoChangedPaths += 1;
+    assertEquals(previousSkippedStreams + 0, skippedStreams.getCount());
+    previousSkippedStreams += 0;
+
+    //Sync the path(/dir0/file0) again, the attempt is not forced so should be skipped
+    syncScheme = new LockingScheme(new AlluxioURI(path), InodeTree.LockPattern.READ, false);
+    syncStream = new InodeSyncStream(syncScheme, mFileSystemMaster, RpcContext.NOOP,
+        DescendantType.NONE, ListStatusContext.defaults().getOptions().getCommonOptions(),
+        false, false, false, false);
+    syncStream.sync();
+
+    assertEquals(previousStreamCount + 1, streamCount.getCount());
+    previousStreamCount += 1;
+    assertEquals(previousSucceededStreams + 0, succeededStreams.getCount());
+    previousSucceededStreams += 0;
+    assertEquals(previousSucceededPaths + 0, succeededPaths.getCount());
+    previousSucceededPaths += 0;
+    assertEquals(previousFailedStreams + 0, failedStreams.getCount());
+    previousFailedStreams += 0;
+    assertEquals(previousFailedPaths + 0, failedPaths.getCount());
+    previousFailedPaths += 0;
+    assertEquals(previousNoChangedPaths + 0, noChangedPaths.getCount());
+    previousNoChangedPaths += 0;
+    assertEquals(previousSkippedStreams + 1, skippedStreams.getCount());
+    previousSkippedStreams += 1;
+
+    // The sync should succeed and the paths(/dir0/file0) should be removed from inodeTree
+    // because inodeTree contains the paths when ufs throwing IOException
+    mUfs.mIsThrowIOException = true;
+    syncScheme = new LockingScheme(new AlluxioURI(path), InodeTree.LockPattern.READ, true);
+    syncStream = new InodeSyncStream(syncScheme, mFileSystemMaster, RpcContext.NOOP,
+        DescendantType.NONE, ListStatusContext.defaults().getOptions().getCommonOptions(),
+        false, true, false, false);
+    syncStream.sync();
+
+    assertEquals(previousStreamCount + 1, streamCount.getCount());
+    previousStreamCount += 1;
+    assertEquals(previousSucceededStreams + 1, succeededStreams.getCount());
+    previousSucceededStreams += 1;
+    assertEquals(previousSucceededPaths + 1, succeededPaths.getCount());
+    previousSucceededPaths += 1;
```

Comment:
why is this a succeeded path?

Reply:
If the target is existed in InodeTree and InodeSyncStream can't get the path status form UFS, the path will be removed from InodeTree and this sync will be considered as succeed.

Reply:
https://github.com/Alluxio/alluxio/blob/0cc777718e9661f85a1568b51f23da44bbdb197b/core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java#L296-L303